File: app/src/main/java/dev/iotarho/artplace/app/ui/artworks/ArtworksViewModelFactory.java
package dev.iotarho.artplace.app.ui.artworks;

import androidx.annotation.NonNull;
import androidx.lifecycle.ViewModel;
import androidx.lifecycle.ViewModelProvider;

import dev.iotarho.artplace.app.repository.ArtsyRepository;

public class ArtworksViewModelFactory implements ViewModelProvider.Factory {

    private ArtsyRepository artsyRepository;

    public ArtworksViewModelFactory(ArtsyRepository artsyRepository) {
        this.artsyRepository = artsyRepository;
    }

    @SuppressWarnings("unchecked")
    @NonNull
    @Override
    public <T extends ViewModel> T create(@NonNull Class<T> modelClass) {
        if (modelClass.isAssignableFrom(ArtworksViewModel.class)) {
            return (T) new ArtworksViewModel(artsyRepository);
        }
        throw new IllegalArgumentException("Unknown ViewModel class.");
    }
}

File: app/src/main/java/dev/iotarho/artplace/app/ui/artworks/ArtworksViewModelFactory.kt
package dev.iotarho.artplace.app.ui.artworks

import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider
import dev.iotarho.artplace.app.repository.ArtsyRepository

class ArtworksViewModelFactory(private val artsyRepository: ArtsyRepository) : ViewModelProvider.Factory {

    override fun <T : ViewModel> create(modelClass: Class<T>): T {
        if (modelClass.isAssignableFrom(ArtworksViewModel::class.java)) {
            return ArtworksViewModel(artsyRepository) as T
        }
        throw IllegalArgumentException("Unknown ViewModel class.")
    }
}